src_features/signMessageEIP712/format_hash_field_type.c

```diff
@@ -0,0 +1,116 @@
+#ifdef HAVE_EIP712_FULL_SUPPORT
+
+#include "format_hash_field_type.h"
+#include "mem.h"
+#include "mem_utils.h"
+#include "commands_712.h"
+#include "hash_bytes.h"
+#include "apdu_constants.h"  // APDU response codes
+#include "typed_data.h"
+
+/**
+ * Format & hash a struct field typesize
+ *
+ * @param[in] field_ptr pointer to the struct field
+ * @param[in] hash_ctx pointer to the hashing context
+ * @return whether the formatting & hashing were successful or not
+ */
+static bool format_hash_field_type_size(const void *const field_ptr, cx_hash_t *hash_ctx) {
+    uint16_t field_size;
+    char *uint_str_ptr;
+    uint8_t uint_str_len;
+
+    field_size = get_struct_field_typesize(field_ptr);
+    switch (struct_field_type(field_ptr)) {
+        case TYPE_SOL_INT:
+        case TYPE_SOL_UINT:
+            field_size *= 8;  // bytes -> bits
+            break;
+        case TYPE_SOL_BYTES_FIX:
+            break;
+        default:
+            // should not be in here :^)
+            apdu_response_code = APDU_RESPONSE_INVALID_DATA;
+            return false;
+    }
+    if ((uint_str_ptr = mem_alloc_and_format_uint(field_size, &uint_str_len)) == NULL) {
```

Comment:
I would prefer no assignation in if

Reply:
Fixed in [174d357](https://github.com/LedgerHQ/app-ethereum/pull/327/commits/174d35782f739d2da5beeec2006831fae79b4d30)